gobblin-runtime/src/main/java/org/apache/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -737,11 +738,19 @@ private static FileSystem buildFileSystem(Properties jobProps, Configuration con
     // A list of WorkUnits (flattened for MultiWorkUnits) to be run by this mapper
     private final List<WorkUnit> workUnits = Lists.newArrayList();
 
+    private AutomaticTroubleshooter troubleshooter;
+    private IssueRepository issueRepository;
+
     @Override
     protected void setup(Context context) {
       final State gobblinJobState = HadoopUtils.getStateFromConf(context.getConfiguration());
       TaskAttemptID taskAttemptID = context.getTaskAttemptID();
 
+      issueRepository = new InMemoryIssueRepository();
+      troubleshooter = new AutomaticTroubleshooter(issueRepository);
+
+      troubleshooter.startAsync().awaitRunning();
```

Comment:
This one should probably have a 1 minute timeout as well to be consistent with the other starts/stops of the troubleshooter.

Reply:
Replaced with synchronous calls.